src/test/kotlin/de/bund/digitalservice/useid/widget/WidgetControllerIntegrationTest.kt

```diff
@@ -129,9 +130,58 @@ class WidgetControllerIntegrationTest(@Autowired val webTestClient: WebTestClien
             .expectBody()
             .returnResult()
 
-        val body = result.responseBody?.decodeToString()
-        assertThat(body, containsString(widgetProperties.errorView.fallback.localization.errorTitle))
-        assertThat(body, containsString("class=\"container fallback\""))
+        val responseBody: String? = result.responseBody?.decodeToString()
+        val parsedResponseBody = Jsoup.parse(responseBody!!)
+
+        val containerFallback = parsedResponseBody.body().firstElementChild()?.className()
+        val hasValidFallbackClassName = containsString("container fallback")
+
+        assertThat(containerFallback, hasValidFallbackClassName)
+
+        val errorTitle = parsedResponseBody.getElementsByClass("error_title").text()
+        val hasValidErrorTitle = containsString(widgetProperties.errorView.fallback.localization.errorTitle)
+
+        assertThat(errorTitle, hasValidErrorTitle)
+    }
+
+    @Test
+    fun `fallback page should encode tcTokenURL param for identification button when the query param is passed`() {
+        val encodedTcTokenUrl = "https%3A%2F%2Fwww.foo.bar"
+        val result = webTestClient
+            .get()
+            .uri("/$FALLBACK_PAGE?tcTokenURL=$encodedTcTokenUrl")
+            .exchange()
+            .expectStatus().isOk
+            .expectBody()
+            .returnResult()
+
+        val responseBody: String? = result.responseBody?.decodeToString()
+        val parsedResponseBody = Jsoup.parse(responseBody!!)
+
+        val eidClientButton = parsedResponseBody.getElementById("eid-client-button")?.attr("href")
+        val hasCorrectUrl = containsString("bundesident://127.0.0.1:24727/eID-Client?tcTokenURL=https%253A%252F%252Fwww.foo.bar")
+
+        assertThat(eidClientButton, hasCorrectUrl)
+    }
+
+    @Test
+    fun `fallback page should encode tcTokenURL param for identification button when the query param is not encoded`() {
+        val decodedTcTokenUrl = "https://www.foo.bar"
```

Comment:
maybe this does not make any sense, since Spring decodes query param automatically